projects/common/src/time/page-time-range-preference.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@angular/core';
+import { isNil } from 'lodash-es';
+import { Observable } from 'rxjs';
+import { map, shareReplay, take } from 'rxjs/operators';
+import { NavigationService } from '../navigation/navigation.service';
+import { PreferenceService, StorageType } from '../preference/preference.service';
+import { TimeRange } from './time-range';
+import { TimeRangeService } from './time-range.service';
+
+@Injectable({ providedIn: 'root' })
+export class PageTimeRangePreferenceService {
+  private static readonly STORAGE_TYPE: StorageType = StorageType.Session;
+  private static readonly TIME_RANGE_PREFERENCE_KEY: string = 'page-time-range';
+
+  private readonly pageTimeRangeStringDictionary$: Observable<PageTimeRangeStringDictionary>;
+
+  public constructor(
+    private readonly preferenceService: PreferenceService,
+    private readonly timeRangeService: TimeRangeService,
+    private readonly navigationService: NavigationService
+  ) {
+    this.pageTimeRangeStringDictionary$ = this.buildPageTimeRangeObservable();
+  }
+
+  public getTimeRangePreferenceForPage(path: string): Observable<TimeRange> {
+    return this.pageTimeRangeStringDictionary$.pipe(
+      map(pageTimeRangeStringDictionary => {
+        if (isNil(pageTimeRangeStringDictionary[path])) {
+          const timeRangeForPath = this.getDefaultPageTimeRange(path);
+          this.setPreferenceServicePageTimeRange(pageTimeRangeStringDictionary ?? {}, path, timeRangeForPath);
```

Comment:
We probably shouldn't persist the default - only if the user picks one.

Reply:
fixe